### PR TITLE
HL-84: Attachments layout

### PR DESCRIPTION
**Pull request by melniiv**
## Description :sparkles:
Attachments step (3) layout is implemented with conditional rendering logic of attachment sections based on benefit type and apprenticeship selections.

### CODE REVIEW

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  showMessage,
+}) => {
+  const { t, translationsBase, attachments } = useAttachmentsList();
+  return (
+    <$Container>
+      <$Heading>
+        {t(`${translationsBase}.types.${camelCase(attachmentType)}.title`)}
+      </$Heading>
+      {attachments.length > 0 ? (
+        <>
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:28]: This fragment tag is not redundant here.

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          <AttachmentItem
+            id="testId"
+            name="palkkatukipaatos-teemurantamaki.PDF"
+            removeText={t(`${translationsBase}.remove`)}
+            onClick={noop}
+            onRemove={noop}
+          />
+        </>
+      ) : (
+        <>
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:38]: And here

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          {showMessage && (
+            <$Message>
+              {t(
+                `${translationsBase}.types.${camelCase(attachmentType)}.message`
+              )}
+            </$Message>
+          )}
+        </>
+      )}
+      <$PrimaryButton style={{ width: 'auto' }} iconLeft={<IconPlus />}>
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:48]: [HDS support full-width buttons](https://hds.hel.fi/storybook/react/?path=/docs/components-button--full-width), you could just pass the `fullWidth` prop with a `true` value.

File: frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+export const $Heading = styled.h1`
+  font-size: ${(props) => props.theme.fontSize.heading.m};
+  font-weight: 500;
+  margin: 0;
+`;
+
+export const $Description = styled.p`
+  font-size: ${(props) => props.theme.fontSize.heading.s};
+  font-weight: normal;
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts:19]: Isn't this the default value?

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/attachmentItem/AttachmentsItem.sc.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  border-bottom: 1px dashed ${(props) => props.theme.colors.coatOfArms};
+  padding: ${(props) => props.theme.spacing.xs};
+  cursor: pointer;
+`;
+
+export const $ActionContainer = styled.span`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  cursor: pointer;
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/attachmentItem/AttachmentsItem.sc.ts:24]: If the container itself is clickable, why don't we move this CSS property to`$Container`?

File: frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  line-height: ${(props) => props.theme.lineHeight.l};
+`;
+
+export const $WarningContainer = styled.div`
+  display: flex;
+  align-items: center;
+  flex: 1 0 30%;
+  box-sizing: border-box;
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts:27]: Redefining box-sizing for elements separately, as opposite to define it globally for the whole project, may cause confusion in the long term. 
But maybe this is a discussion for later.

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
Left a few comments.

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/attachmentItem/AttachmentsItem.sc.ts (hunk 5 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/attachmentItem/AttachmentsItem.sc.ts:24]: its not, as there is a space between the attachment and remove button, which is not clickable

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx (hunk 3 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:48]: we want here exactly opposite, fullwidth is width 100% i guess

File: frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts (hunk 6 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts:27]: yea, i should remove it. good point

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx (hunk 1 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:28]: yes, i ll remove

File: frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx (hunk 2 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx:38]: here its needed

File: frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts (hunk 4 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/attachmentsIngress/AttachmentsIngress.sc.ts:19]: fixed

**Review by Nizar-Rahme — APPROVED** [review]

**Pull request merged**